src/utils.ts

```diff
@@ -142,3 +143,106 @@ export function voidFn() {}
 export function assertUnreachable(x: never): never {
   throw new Error("Didn't expect to get here");
 }
+
+/**
+ * A custom React Hook for calculating whether a thumb overlaps
+ * another and whether labels could/should merge.
+ * @param rangeRef - React ref value of Range component
+ * @param values - current Range values Array
+ * @param index - thumb index
+ * @param separator - string to separate thumb values
+ */
+export const useThumbOverlap = (
+  rangeRef: IRangeRef | undefined,
+  values: number[],
+  index: number,
+  separator = '-',
+) => {
+  const [style, setStyle] = useState<{ [key: string]: any }>({});
+  const [labelValue, setLabelValue] = useState(values[index].toFixed(1));
+
+  useEffect(() => {
+    if (rangeRef) {
+      let overlaps = [];
+      let labelValue = values[index].toFixed(1);
+      const offsets = rangeRef.getOffsets();
+      const THUMB_SIZE = rangeRef.getThumbs()[0].offsetWidth;
```

Comment:
I would check if `rangeRef.getThumbs().length > 0`. We throw a warning already but it would nice to prevent the crash.